tests/metrics_test_utils.go

```diff
@@ -0,0 +1,59 @@
+package tests
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"regexp"
+	"strconv"
+
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+var regexpForMetrics = map[string]string{
+	"total_rejected_vms": `total_rejected_vms ([0-9]+)`,
+}
+
+func intMetricValue(metricName string, metricsPort uint16, pod *v1.Pod) int {
+	cfg, err := config.GetConfig()
+	Expect(err).ToNot(HaveOccurred())
+	coreClient, err := kubernetes.NewForConfig(cfg)
+	Expect(err).ToNot(HaveOccurred())
+
+	portForwarder := NewPortForwarder(cfg, coreClient.CoreV1().RESTClient())
+	conn, _ := portForwarder.Connect(pod, metricsPort)
+	client := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
+				return conn, nil
+			},
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+	resp, err := client.Get(fmt.Sprintf("https://localhost:%d/metrics", metricsPort))
+	Expect(err).To(BeNil(), "Can't get metrics from %s", pod.Name)
+	defer resp.Body.Close()
+	body, _ := ioutil.ReadAll(resp.Body)
+	regexString, ok := regexpForMetricByName(metricName)
+	if !ok {
+		panic(fmt.Sprintf("metricName %s does not have a defined regexp string, please add one to the regexpForMetrics map", metricName))
+	}
+	regex := *regexp.MustCompile(regexString)
+	valueOfMetric := regex.FindSubmatch(body)
+	intValue, _ := strconv.Atoi(string(valueOfMetric[1]))
+	return intValue
+}
+
+func regexpForMetricByName(metricName string) (string, bool) {
+	regexp, ok := regexpForMetrics[metricName]
+	if !ok {
```

Comment:
This `if` is not needed. If an element is not found in `regexpForMetrics`, it returns `"", false`.

Reply:
yep! removed this useless function.